ADAL/src/telemetry/ADTelemetry.m

```diff
@@ -179,6 +177,13 @@ - (void)dispatchEventNow:(NSString*)requestId
     {
         for (ADDefaultDispatcher *dispatcher in _dispatchers)
         {
+            for (NSString *propertyName in [event.propertyMap allKeys]) {
+                BOOL isPii = [[ADTelemetryPiiRules sharedInstance] isPii:propertyName];
+                if (isPii && !self.piiEnabled) {
+                    [event setProperty:propertyName value:@""];
```

Comment:
For bandwidth optimization I'd prefer removing the value for this property completely instead of sending an empty telemetry value.